src/tasks/LibraryBuilder/Templates/autoinit.c

```diff
@@ -0,0 +1,38 @@
+#include <stdint.h>
+#include <stdio.h>
+#include <mono/jit/jit.h>
+#include <mono/jit/mono-private-unstable.h>
+
+/*
+    Note that this will expand into auto-initialization the runtime, but for now
+    it makes sure to keep the monovm_initialize and monovm_runtimeconfig_initialize symbols
+*/
+
+void keep_init()
+{
+    const char* appctx_keys[3];
+    appctx_keys[0] = "RUNTIME_IDENTIFIER";
+    appctx_keys[1] = "APP_CONTEXT_BASE_DIRECTORY";
+    appctx_keys[2] = "System.TimeZoneInfo.LocalDateTimeOffset";
+
+	const char* appctx_values[3];
+    appctx_values[0] = "RUNTIME_IDENTIFIER";
+    appctx_values[1] = "APP_CONTEXT_BASE_DIRECTORY";
+    appctx_values[2] = "System.TimeZoneInfo.LocalDateTimeOffset";
```

Comment:
this isn't setting any real values, just strings?

Reply:
right, first this is temporary until https://github.com/dotnet/runtime/pull/82253 lands.  It's here because on Android `monovm_initialize` and `monovm_runtimeconfig_initialize` weren't being preserved as symbols and so a test app would not build.  The library will not export these functions at all.